content/english/debugging/04-identify-the-problem3.md

```diff
@@ -0,0 +1,171 @@
+
+If you want to have your code continue running until the next breakpoint, you can use the `continue` (or `c`) command. Finally, to delete a breakpoint, you can use the `delete` (or `d`) command followed by the breakpoint number, which you see when you set the breakpoint.
+
+Another useful command to know is `bt` or `backtrace`. This shows you the calling stack, or which functions were called in order to get to the line of code the program is halted on. This can be helpful in tracking execution order and finding control flow bugs. It also works well to find out how a program crashed by seeing the functions that led up to the crash. 
+
+There are plenty of commands GDB offers. We recommend using a cheatsheeet that you can find online. These cheat sheets work:
+
+* [Reference 1](https://gist.github.com/rkubik/b96c23bd8ed58333de37f2b8cd052c30)
+* [Reference 2](https://cs.brown.edu/courses/cs033/docs/guides/gdb.pdf)
+
+### Planning Where to Place Breakpoints
+
+For this example, where is all the sorting actually taking place?
+{{% expand "*Click to show answer*" %}}
+The sorting takes place within the `for` loop. Here, the elements are being moved around. 
+
+Another place the sorting takes place is in the base case - if there are two elements, the function simply swaps the values if they are not sorted. Does the code reflect that?
+{{% /expand %}}
+<br/>
+You should now insert a breakpoint where much of the sorting takes place (with the `break` command). Next, enter the `run` command, and say 'yes' if GDB asks you to restart the program. The program should stop at your breakpoint.
+
+Here's where the frustrating part happens - you need to find the faulty variables that cause your program to not work as expected. What variables are changing within the loop?
+
+{{% expand "*Click to show answer*" %}}
+`i`, `pivotIdx`, and `tmp` change (technically `tmp` is declared inside the loop though). 
+{{% /expand %}}
+<br/>
+You can make GDB display these variables every time the debugger pauses by using the `disp` command. Run these two commands:
+
+```bash
+disp i
+disp pivotIdx
+```
+
+Now, whenever the debugger stops, it'll display the values of `i` and `pivotIdx`.
+
+At this stage, it is crucial to understand what the implementation is doing. Refer to the picture above to get an idea of the process, then try to match it with the code. Next, run a few iterations of the loop, watching the indices update. If you need a hint, open the expander below.
+
+{{% expand "*Click to show hint*"%}}
+When should `i` be updated - in the loop header, or under certain conditions within the loop body?
+{{% /expand %}}
+<br/>
+
+Note that all debuggers should have the same basic concepts: they allow you to step through your code as it runs, in real time. You should almost always use debuggers over print statements. You won't regret it!
+
+## Another Type of Bug: Memory Errors

```

Comment:
To make the content more consumable, perhaps split this Valgrind section into a new page